File: 3-Longest Substring Without Repeating Characters/LengthOfLongestSubstring.py
def lengthOfLongestSubstring(s : str) -> int :
    hashmap = {}
    l = 0
    rslt = 0
    
    for r in range(len(s)):
        if s[r] in hashmap and hashmap[s[r]] + 1 > l:
            l = hashmap[s[r]] + 1
        hashmap[s[r]] = r
        
        rslt = max(rslt,r - l + 1)
        
    return rslt


print(lengthOfLongestSubstring("abcabcbb"))

File: 3-Longest Substring Without Repeating Characters/LengthOfLongestSubstring.java

import java.util.HashMap;

public class LengthOfLongestSubstring {

    static public int solution(String s) {
        HashMap<Character,Integer> map = new HashMap<>();
        int rslt,left;
        left = rslt = 0;

        for(int right = 0 ; right < s.length() ; right++) {
            if(map.containsKey(s.charAt(right)) && map.get(s.charAt(right)) + 1 > left) {
                left = map.get(s.charAt(right)) + 1;
            }
            map.put(s.charAt(right), right);

            rslt = Math.max(rslt, right - left + 1);
        }
        return rslt;
    }
    public static void main(String[] args) {
        System.out.println(solution("abcabcbb"));   
    }
}
